flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -684,12 +713,139 @@ public void alterPartitionColumnStatistics(ObjectPath tablePath, CatalogPartitio
 
   @Override
   public List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
+      throws TableNotExistException, TableNotPartitionedException, CatalogException, PartitionSpecInvalidException {
+    Preconditions.checkNotNull(tablePath, "Table path cannot be null");
+    Preconditions.checkNotNull(partitionSpec, "CatalogPartitionSpec cannot be null");
+
+    Table table = loadIcebergTable(tablePath);
+    ensurePartitionedTable(tablePath, table);
+    Optional<PartitionSpec> matchingPartitionSpec = getMatchingPartitionSpec(partitionSpec, table.specs());
+    if (!matchingPartitionSpec.isPresent()) {
+      throw new PartitionSpecInvalidException(getName(),
+          table.specs().values().stream()
+              .map(spec -> spec.fields().stream()
+                  .map(PartitionField::name)
+                  .collect(Collectors.joining(",", "{", "}")))
+              .collect(Collectors.toList()),
```

Comment:
This seems was trying to encode all the iceberg partition spec into a `List<String>` to `partitionKeys` ?  I don't think it's correct because the `PartitionSpecInvalidException`'s `partitionKeys` are partition key names that parsed from a specific partition spec, it's usually the latest iceberg partition specification spec, right ?

Reply:
Yes, `partitionKeys` should be a list of partition key names. The current implementation is a list of all partition spec keys. For example, a table with two specs may display something like [{key1, key2}, {key1, key2, key3}]. Should we merge partition key names from all partition specs as `partitionKeys`, like [key1, key2, key3]? Because this exception is thrown after looking for all partition specs but still not found a matching spec. Or should we use other exception for this situation? But I do not see any suitable exception class in `org.apache.flink.table.catalog.exceptions`.